### DISCUSSION TITLE
Product Version

### DISCUSSION THREAD

**Discussion by KMarois** [wixtoolset/issues#7195] [Questions]
I am brand new to Wix, so bear with me...

I am using Wix 3 to build a Proof of Concept installer to install a WPF application. 

At build time, I want to retrieve the Product Version from the WPF app's exe. I'm trying this, but it doesn't compile:

```
<?define PRODUCT=$(var.WixDemo.TargetName)?>
<?define MAINEXE=$(var.WixDemo.TargetFileName)?>
<?define VERSION=""?> <============= WHAT GOES HERE??
	
<Product Id="*" 
			Name="$(var.PRODUCT)"
			Version="$(var.VERSION)"
			Language="1033" 
			Manufacturer="Acme Widgets" 
			UpgradeCode="{0C6D0563-C288-4217-90ED-904DCF799609}">

```
I'm using Heat to create import the files, and if I hard code the generated GUI from the EXE file, that does work:
`<?define VERSION="!(bind.fileVersion.filD7191EADF27B9A1BF81DFB89BFDB8E4F)"?>`

but that's not a solution because the GUI changes from time to time.

There doesn't seem to be a variable for Version in the supported variables table:
https://wixtoolset.org/docs/v3/votive/votive_project_references/

Thanks

**Comment by robmen** [comment]
I just did a Deployment Dojo episode ["Variables End to End."](https://www.youtube.com/live/d4DEPW-RO30?feature=share) that walks through the different variable types. It uses WiX v4 but the same concepts apply to WiX v3. If you're new, it should be helpful to contextualize all the different variables you're using.

1. I'd recommend dropping the `<?define?>` preprocessor variables to remove some complexity. Just put the bind variables in the `Product`'s `Name`, and `Version` attributes to get started.
2. Harvesting using heat.exe creates challenges like this since it generates complex ids. Some people use XSLTs to transform the heat output to hammer it into a shape they like.

Personally, I don't like messing around with heat and use [FireGiant's Harvesting from the WiX Expansion Pack](https://www.firegiant.com/wix/wep-documentation/harvesting/). If you use that harvesting solution, then you can set the `KeyIdentifier` attribute on the `HarvestProject` to get a friendly identifier for the WPF exe. [Using the example in the documentation](https://www.firegiant.com/wix/wep-documentation/harvesting/key-identifier/), you're bind variable would look like `!(bind.fileVersion.MyConsoleApp)`.

**Reply by MaroisConsulting** [reply]
OK thanks, I'll take a look

**Reply by MaroisConsulting** [reply]
On a side note, is tthere any UI available for building installers using Wix?

**Reply by robmen** [reply]
There are a few out there. I have this [really old list](https://robmensching.com/blog/posts/2007/11/20/wix-editors/). More are coming in the future as well.

**Reply by KMarois** [reply]
Rob, I want to look at the Advanced Harvesting option, but I can't seem to find the FgwepExtension.wixext.dll referenced on that page. There's no NuGet reference. What am I missing?

**Reply by robmen** [reply]
That "Advanced Harvesting" is part of the [WiX Expansion Pack (WEP)](https://www.firegiant.com/products/wix-expansion-pack/). You can get a trial license by [creating a FireGiant Account](https://account.firegiant.com/licenses/).

For WiX v4 (-rc.3 if all goes to plan), the WEP will be available as a NuGet package (and thus easier to acquire).

**Reply by KMarois** [reply]
So is there any way to do the harvestingoption without buying a license?

**Reply by robmen** [reply]
The FireGiant license file is designed to be easy to integrate into the build. You can just check the file in any folder equal to or above your .wixproj. There are other options documented on the [FireGiant Licensing page](https://www.firegiant.com/wix/wep-documentation/getting-started/firegiant-licensing/).
